### PR TITLE
fix: address reviews

### PR DESCRIPTION
**Pull request by gurdeep330**
# For authors

## Description

In this PR, I'd implement feature requests/fix bug reports received from users:
- **`load_biomodel` utility:** Extend support for BioModels starting with `MODEL*`.  
- **`search_models` tool:**  
  - Replace markdown tables with dataframes.  
  - Increase result limit to 100 (upper limit in basico).  
- **Model handling:** Fix support for models that don't return parameters (e.g., BioModel 10).
- **Plotly charts:** Improve x/y axis labels to be more descriptive, including units.  
- **`steady_state` tool:** Display steady-state table instead of just a tool message.  
- **Starter message:** Add sample question on parameter scanning and annotations.  

## Type of change
Please delete options that are not relevant.
- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?
Please describe the tests you conducted to verify your changes. These may involve creating new test scripts or updating existing ones.
- [X] Added new test(s) in the ```tests``` folder
- [ ] Added new function(s) to an existing test(s) (e.g.: ```tests/testX.py```)
- [ ] No new tests added (Please explain the rationale in this case)

## Checklist
- [X] My code follows the style guidelines mentioned in the Code/DevOps guides
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation (e.g. MkDocs)
- [X] My changes generate no new warnings
- [X] I have added or updated tests (in the ```tests``` folder) that prove my fix is effective or that my feature works
- [X] New and existing tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

# For reviewers

## Checklist pre-approval
- [ ] Is there enough documentation?  
- [ ] If a new feature has been added, or a bug fixed, has a test been added to confirm good behavior?
- [ ] Does the test(s) successfully test edge/corner cases?
- [ ] Does the PR pass the tests? (if the repository has continuous integration)

## Checklist post-approval
- [ ] Does this PR merge ```develop``` into ```main```? If so, please make sure to add a prefix (feat/fix/chore) and/or a suffix BREAKING CHANGE (if it's a major release) to your commit message.
- [ ] Does this PR close an issue? If so, please make sure to descriptively close this issue when the PR is merged.

## Checklist post-merge
- [ ] When you approve of the PR, merge and close it (Read this [article](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/about-merge-methods-on-github) to know about different merge methods on GitHub)
- [ ] Did this PR merge ```develop``` into ```main``` and is it suppose to run an automated release workflow (if applicable)? If so, please make sure to check under the "Actions" tab to see if the workflow has been initiated, and return later to verify that it has completed successfully.

### CODE REVIEW

File: aiagents4pharma/talk2biomodels/tools/parameter_scan.py [hunk 1]
```diff
@@ -22,6 +22,21 @@
 logging.basicConfig(level=logging.INFO)
 logger = logging.getLogger(__name__)
 
+def get_model_units(model_object):
```
Comment (dmccloskey) [inline aiagents4pharma/talk2biomodels/tools/parameter_scan.py:25]: I believe this is the second time this exact same function has popped up. I would recommend consolidating the different versions into a single function that can be incorporated by the different modules as needed.

File: app/frontend/utils/streamlit_utils.py [hunk 2]
```diff
@@ -310,43 +324,77 @@ def get_response(app, st, prompt):
...
+                toggle_state=True,
                 save_toggle=True)
             # Display the table
             render_table(
                 df_selected,
                 key="dataframe_"+uniq_msg_id,
-                # tool_name=msg.name,
-                # tool_call_id=msg.tool_call_id,
                 save_table=True)
+        elif msg.name == "search_models":
```
[the remark this answers is not in the document]
Comment (gurdeep330) [reply app/frontend/utils/streamlit_utils.py:356]: Can you share the link again? The one you shared above takes me to the top of 'Files Changed' tab.
Anyway, I am trying to consolidate tools that require chart and table rendering using the function `render_table_plotly` in app/frontend/utils/streamlit_utils.py. This was previously OK when any tool that required chart rendering also required table rendering. But things are changing a little now. Some new tools require just table rendering (e.g. search_models or steady-state).  It’s still evolving as I refine the exact requirements. I’ll revisit and clean it up after a few more PRs?

File: aiagents4pharma/talk2biomodels/tools/parameter_scan.py (hunk 1 above)
Comment (gurdeep330) [reply aiagents4pharma/talk2biomodels/tools/parameter_scan.py]: That's a good catch! I moved it to `utils.py`.

File: aiagents4pharma/talk2biomodels/tools/custom_plotter.py [hunk 3]
```diff
@@ -1,25 +1,95 @@
...
+        """
+        relevant_species: Union[None, List[Literal[*species_names]]] = Field(
+                description="This is a list of species based on the user question."
+                "It is restricted to the species available in the simulation results."
+                "If no species is relevant, set this attribute to None."
+                "If the user asks for very specific species (for example, using the"
+                "keyword `only` in the question), set this attribute to correspond "
+                "to the species available in the simulation results, otherwise set it to None."
+                )
+    system = '''You are custom plotter tool. You can extract species from the given
```
[the remark this answers is not in the document]
Comment (gurdeep330) [reply aiagents4pharma/talk2biomodels/tools/custom_plotter.py]: Let's move it to configs. I am sure in the future, this tool will get massively updated with plenty of configs :-)

File: aiagents4pharma/talk2biomodels/tools/custom_plotter.py [hunk 4]
```diff
@@ -1,25 +1,95 @@
...
+                keyword `only` or `exactly` in the question), set this attribute to
+                correspond strictly to the species available in the simulation results,
+                otherwise set it to None.'''
+    # Create an instance of the LLM model
+    logging.log(logging.INFO, "LLM model: %s", state['llm_model'])
+    llm = state['llm_model']
+    llm_with_structured_output = llm.with_structured_output(CustomHeader)
+    prompt = ChatPromptTemplate.from_messages([("system", system), ("human", "{input}")])
+    few_shot_structured_llm = prompt | llm_with_structured_output
+    # return llm_with_structured_output.invoke(question)
```
[the remark this answers is not in the document]
Comment (gurdeep330) [reply aiagents4pharma/talk2biomodels/tools/custom_plotter.py]: Nope. Thanks for pointing it out!

**Comment by gurdeep330** [comment]
@dmccloskey Thank you for the review and all the suggestions. Let me know if anything is still missing.

File: app/frontend/utils/streamlit_utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dmccloskey) [reply app/frontend/utils/streamlit_utils.py:356]: Sorry for the bad link. Gotcha. Let's revisit once things stabilize on this part then.

**Review by dmccloskey — APPROVED** [review]
All clear 😊
